src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextProvider.cs

```diff
@@ -79,10 +79,15 @@ public async Task ReleaseProjectContextAsync(IWorkspaceProjectContextAccessor ac
 
                 try
                 {
-                    // Update LastDesignTimeBuildSucceeded within a batch to avoid thread pool starvation.
+                    // Update additional properties within a batch to avoid thread pool starvation.
                     // https://github.com/dotnet/project-system/issues/8027
 
                     context.LastDesignTimeBuildSucceeded = false;  // By default, turn off diagnostics until the first design time build succeeds for this project.
+
+                    // Pass along any early approximation we have of the command line options
+#pragma warning disable CS0618 // This was obsoleted in favor of the one that takes an array, but here just the string is easier; we'll un-Obsolete this API
```

Comment:
Because I'm using a property here rather than an item group, this overload that we obsoleted in favor of the one that takes an item group is easier to use. It wasn't clear to me how to grab an item group from evaluation in the rule and if that's easy to change I'm happy to do so; otherwise we'll just use this overload.

The reason this was obsoleted was because this string can be _very_ large in the case of the fully evaluated one (since it contains references and source file paths), and was ending up in the large object heap. But in this case, this string isn't nearly as large.

Reply:
What does the string look like in this case?

Reply:
@tmeschter: good question! The Roslyn side of the change is https://github.com/dotnet/roslyn/pull/60855, so you are basically getting `-langver:... -define:...`